src/main.py

```diff
@@ -0,0 +1,201 @@
+import yaml
+from telebot import TeleBot
+from telebot.types import BotCommand, Message
+from data_collector.camera_utils import get_camera, get_camera_image
+from data_collector.gdrive import upload_photo
+from enum import StrEnum
+import time
+from io import BytesIO
+from PIL.Image import Image
+import threading
+
+
+class CaptureStatus(StrEnum):
+    NOT_STARTED = "not started"
+    RUNNING = "running"
+
+
+current_status: CaptureStatus = CaptureStatus.NOT_STARTED
+capture_thread: threading.Thread | None = None
+capture_thread_duration: int = -1
+
+
+def _peek(message: Message, bot: TeleBot) -> None:
+    """
+    This function sends the current camera view
+    :param message: The message object
+    :param bot: The bot object
+    """
+    peek: Image | None = get_camera_image()
+    if peek is not None:
+        image_bytes = BytesIO()
+        image_bytes.name = 'peek.jpeg'
+        peek.save(image_bytes, 'JPEG')
+        image_bytes.seek(0)
+        bot.send_photo(message.chat.id, photo=image_bytes, message_thread_id=message.message_thread_id)
+    else:
+        bot.send_message(message.chat.id, "Unable to capture image",
+                         message_thread_id=message.message_thread_id)
+
+
+def capture_job(duration: int) -> None:
+    """
+    This function captures images from the camera every 30 seconds. The process will run for the specified duration.
+    If duration is 0, the process will run indefinitely
+    """
+    global current_status, capture_thread_duration
+    internal_counter: int = 0
+    endless = duration == 0
+    duration = duration * 60
+    try:
+        while current_status != CaptureStatus.NOT_STARTED:
+            if internal_counter == 0:
+                image: Image | None = get_camera_image()
+                upload_photo(image)
+                if not endless:
+                    duration -= 30
+                    if duration <= 0:
+                        print("Capture process finished")
+                        current_status = CaptureStatus.NOT_STARTED
+                        capture_thread_duration = -1
+                        break
+            # This module should trigger the capture process every 30 seconds
+            internal_counter = (internal_counter + 1) % 60
+            time.sleep(0.5)
+
+    except KeyboardInterrupt:
+        pass
+
+
+def _start_capture_process(message: Message, bot: TeleBot) -> None:
+    """
+    This function starts the capture service for the next 30 minutes
+
+    :param message: The message object
+    :param bot: The bot object
+    """
+    global current_status, capture_thread, capture_thread_duration
+    command = message.text.split()
+    if len(command) == 1 or not command[1].isdigit():
```

Comment:
non so se ha senso mettere una sorta di null-check tipo `len(command) > 0`

Reply:
In quel contesto ci va solo se il comando /capture viene rilevato dalla libreria telebot, ergo uno split su "/capture <la-qualsiasi>" fornirà sempre un array di almeno 1 elemento. Questo controllo è superfluo